src/wrapped.rs

```diff
@@ -0,0 +1,112 @@
+#![allow(missing_docs)]
+
+use {StringInterner, Symbol, Iter, Values};
+use std::collections::hash_map::RandomState;
+use std::hash::BuildHasher;
+use std::ops::Deref;
+
+#[derive(Copy, Clone, Debug)]
+pub struct PooledStr<'pool, Sym: Symbol + 'pool = usize, H: BuildHasher + 'pool = RandomState> {
+	pool: &'pool StringPool<Sym, H>,
+	sym: Sym,
+}
+
+impl<'pool, Sym: Symbol + 'pool, H: BuildHasher + 'pool> Eq for PooledStr<'pool, Sym, H> {}
+impl<'pool, Sym: Symbol + 'pool, H: BuildHasher + 'pool> PartialEq<Self> for PooledStr<'pool, Sym, H> {
+	fn eq(&self, other: &Self) -> bool {
+		self.sym == other.sym && ::std::ptr::eq(self.pool, other.pool)
+	}
+}
+
+impl<'pool, Sym: Symbol + 'pool, H: BuildHasher + 'pool> Deref for PooledStr<'pool, Sym, H> {
+	type Target = str;
+	fn deref(&self) -> &str {
+		// in the future, maybe use resolve_unchecked
+		PooledStr::resolve(self)
+	}
+}
+
+impl<'pool, Sym: Symbol + 'pool, H: BuildHasher + 'pool> PooledStr<'pool, Sym, H> {
+	pub fn resolve(this: &Self) -> &str {
+		this.pool.interner.resolve(this.sym)
+			.expect("PooledStr exists without entry in StringPool")
+	}
+
+	pub unsafe fn resolve_unchecked(this: &Self) -> &str {
+		this.pool.interner.resolve_unchecked(this.sym)
+	}
+}
+
+#[derive(Debug)]
+pub struct StringPool<Sym: Symbol = usize, H: BuildHasher = RandomState> {
+	interner: StringInterner<Sym, H>,
+}
+
```

Comment:
A `Default` impl similar to `StringInterner`'s impl for `Default` would be decent, too. :)

Reply:
👍